src/exts/message.py

```diff
@@ -1,30 +1,81 @@
+    @interactions.extension_message_command(
+        name="Create help thread", scope=src.const.METADATA["guild"]
+    )
     async def create_help_thread(self, ctx: interactions.CommandContext):
-        _guild: dict = await self.bot._http.get_guild(int(ctx.guild_id))
-        guild = interactions.Guild(**_guild, _client=self.bot._http)
+        self.targets[ctx.author.id] = ctx.target
+        modal = interactions.Modal(
+            custom_id="help_thread_creation",
+            title="Create help thread",
+            components=[
+                interactions.TextInput(
+                    style=interactions.TextStyleType.SHORT,
+                    custom_id="help_thread_name",
+                    label="Help thread name:",
+                    value=f"[AUTO] Help thread for {ctx.target.author.username}.",
+                    min_length=1,
+                    max_length=100,
+                ),
+                interactions.TextInput(
+                    style=interactions.TextStyleType.PARAGRAPH,
+                    custom_id="edit_content",
+                    label="Edit the content:",
+                    value=ctx.target.content,
+                    min_length=1,
+                    max_length=2000,
+                ),
+            ],
+        )
+        await ctx.popup(modal)
+
+    @interactions.extension_modal("help_thread_creation")
+    async def help_thread_creation_modal(
+        self, ctx: interactions.CommandContext, thread_name: str, content: str
+    ):
+        target: interactions.Message = self.targets.pop(ctx.author.id)
+        # _guild: dict = await self.bot._http.get_guild(int(ctx.guild_id))
+        # guild = interactions.Guild(**_guild, _client=self.bot._http)
 
         # sorry EdVraz, we'll need to manually do it for now until the helper is fixed.
         _thread: dict = await self.bot._http.create_thread(
-            name=f"[AUTO] Help thread for {ctx.target.author.username}.",
+            name=thread_name,
             channel_id=src.const.METADATA["channels"]["help"],
-            thread_type=interactions.ChannelType.GUILD_PUBLIC_THREAD.value
+            thread_type=interactions.ChannelType.GUILD_PUBLIC_THREAD.value,
         )
         thread = interactions.Channel(**_thread, _client=self.bot._http)
 
         await thread.add_member(int(ctx.author.id))
-        await thread.add_member(int(ctx.target.author.id))
-        await thread.send(f"This help thread has been created by {ctx.author.mention} for {ctx.target.author.mention}. Below is the question:\n```\n{ctx.target.content}\n```")
-        await ctx.send(f"{ctx.target.author.mention}, please redirect to {thread.mention} at this time.")
-        await ctx.send(":heavy_check_mark: Thread created.", ephemeral=True)
+        await thread.add_member(int(target.author.id))
+        embed = interactions.Embed(
+            title=content.split("\n")[0][:256]
```

Comment:
What in God's name?

Reply:
the more i read this the more im horrified of it

Reply:
The title is supposed to be the "question" of the person, usually in the first line.
It is limited to 256 characters since that is the limit.

Reply:
I personally feel like this is jarring. It's cool, but let's present this hypothetical:

```
Hey everyone! First off, this is a cool library and al ...

[now insert question]
```

To me this both looks and feels odd to read

Reply:
So how should I do it?

Reply:
I was originally looking at proposing embed fields:

- 2 inline fields for the author of the question, and invoker (helper) with mentions
- a field not inline underneath for "Question"
- a field not inline underneath for "Additional information"

Reply:
And what if the message content is greater than 1024 characters? Since that is the limit of a field. Should I shorten it to 1021 characters and add `...`?

Reply:
Mmh, I think so, yeah. I would've probably truncated with the `...` if I had that issue.

Reply:
So should I scrap the title altogether and have just fields?

Reply:
I *want* to say yes, but I know there's likely people who want a title. Maybe the title should reflect the name of the thread? Not too sure.